### PR TITLE
golangci-lint updates

### PR DESCRIPTION
**Pull request by gilbode**
Upgrade golangci-lint to 1.47.3 and enforce the version in the makefile
Replace deprecated linters deadcode and golint with unused and revive 
Run golangci-lint from the root and fix all errors

### CODE REVIEW

**Review by fredtibbitts — APPROVED** [review]

**Comment by fredtibbitts** [comment]
Looks like there were failures running the linters, a newer version of Go is required?

**Comment by gilbode** [comment]
> Looks like there were failures running the linters, a newer version of Go is required?

Not sure I haven’t been able to figure out what failed yet - need to check

**Review by fredtibbitts — DISMISSED** [review]

**Comment by gilbode** [comment]
Fixed all the issues - eventually went with 1.47.3 instead of 1.49.0 b/c 1.49.0 uses go 1.19 gofmt which wants you to reformat comments but running gofmt w/ older versions of go (like 1.17 which the project is on) won't get rid of the linter errors.

Also - using go install to install golangci-lint has problems b/c the projects requires different go version requirements than verrazzano so abandoned that idea.

File: application-operator/apis/oam/v1alpha1/ingresstrait_webhook_test.go [hunk 1]
```diff
@@ -452,7 +452,9 @@ func TestValidateUpdateAddPath(t *testing.T) {
 // TestValidateUpdateAddHostCollision tests validation of an IngressTrait update where a new host and path is added to an
 // existing trait and that new host and path causes a collision with another existing trait.
 // GIVEN an existing IngressTrait with a specified host and path that is going to be updated
-//   AND another existing IngressTrait with the same different host and path
+//
```
Comment (fredtibbitts) [inline application-operator/apis/oam/v1alpha1/ingresstrait_webhook_test.go:455]: All of these extra blank lines seem like they'll be annoying. Is there any way to control this?

**Review by fredtibbitts — COMMENTED** [review]
I can re-approve but there are merge conflicts that will require another re-approval.

File: application-operator/apis/oam/v1alpha1/ingresstrait_webhook_test.go (hunk 1 above)
Comment (gilbode) [reply application-operator/apis/oam/v1alpha1/ingresstrait_webhook_test.go]: go 1.19 reformats doc comments w/ gofmt and I briefly used a version of golangci-lint that did gofmt based on go 1.19.  for now, I undid all the doc comment changes.  someone will need to make them all when we move to go 1.19 if we want to use the gofmt linter (which i think we probably should).

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**